src/server.js

```diff
@@ -60,22 +68,69 @@ app.use(function *(next) {
 });
 
 /**
+ * Determine the format of the output based on the `format` query parameter.
+ *
+ * The format must be among the formats supported by PhantomJS, else 400
+ * Bad Request is thrown. If no format is provided, the default is used.
+ */
+app.use(function *(next) {
+    const { format = defaultFormat } = this.request.query;
+
+    if (supportedFormats.indexOf(format.toLowerCase()) === -1) {
+        this.throw(`Format ${format} not supported.`, 400);
+    }
+
+    this.type = this.state.format = format;
+    yield next;
+});
+
+/**
+ * Set up the size of the page to render, either the paper size for PDF or
+ * clip rectangle for image formats.
+ */
+app.use(function *(next) {
+    if (this.state.format === 'pdf') {
+        this.state.page.property('paperSize', {
```

Comment:
I *think* this should use `yield` as well, since `page.property` is asynchronous and returns a promise. So to ensure this has been set before we yield to the next middleware a yield of this should be done.